hail/python/hailtop/cleanup_gcr/__main__.py

```diff
@@ -0,0 +1,98 @@
+import sys
+import time
+import logging
+import asyncio
+import aiohttp
+import hailtop.aiogoogle.container as google_container
+
+log = logging.getLogger(__name__)
+
+
+class AsyncIOExecutor:
+    def __init__(self, parallelism):
+        self._semaphore = asyncio.Semaphore(parallelism)
+
+    async def _run(self, fut, aw):
+        async with self._semaphore:
+            try:
+                fut.set_result(await aw)
+            except asyncio.CancelledError:  # pylint: disable=try-except-raise
+                raise
+            except Exception as e:  # pylint: disable=broad-except
+                fut.set_exception(e)
+
+    def submit(self, aw):
+        fut = asyncio.Future()
+        asyncio.ensure_future(self._run(fut, aw))
```

Comment:
I thought I ran into this problem where the awaitable started running. That's why the other async pool functions took either partial functions of the function name, args, kwargs.

Reply:
OK, I didn't quite remember what was going on before, so I dug into this.  First, as far as I can tell, an awaitable doesn't start running until you await it (or wrap it in a task with something like ensure_future).  Here's a test example:

```
$ more foo.py
import asyncio

async def f():
    print('in f')


async def main():
    print('before f')
    aw = f() # don't await
    print('after f, before sleep')
    await asyncio.sleep(3)
    print('after sleep')
    await aw
    print('after await f')

asyncio.run(main())
$ python3 foo.py
before f
after f, before sleep
after sleep
in f
after await f
```

Your PR is here: https://github.com/hail-is/hail/pull/7371

You wrote, "This code doesn't quite work in the case of cancelling."  The problem as far as I can reconstruct is that you can't cancel an awaitable.  My new thing runs all submitted tasks and uses futures to propagate out exceptions.  I think I'm happy with this semantics here, although we might want to revisit it if we move it to utils and/or try to unify our asyncio worker pool abstractions.